src/zenml/integrations/neptune/experiment_trackers/neptune_experiment_tracker.py

```diff
@@ -0,0 +1,70 @@
+from typing import TYPE_CHECKING, Any, Optional, Set, Type, cast
+
+from zenml.config.base_settings import BaseSettings
+from zenml.experiment_trackers.base_experiment_tracker import (
+    BaseExperimentTracker,
+    BaseExperimentTrackerConfig,
+)
+from zenml.integrations.neptune.experiment_trackers.run_state import RunProvider
+from zenml.utils.secret_utils import SecretField
+
+if TYPE_CHECKING:
+    from zenml.config.step_run_info import StepRunInfo
+
+
+class NeptuneExperimentTrackerConfig(BaseExperimentTrackerConfig):
+    project: Optional[str] = None
+    api_token: Optional[str] = SecretField()
+
+
+class NeptuneExperimentTrackerSettings(BaseSettings):
+    tags: Set[str] = set()
+
+
+class NeptuneExperimentTracker(BaseExperimentTracker):
+    """
+    Track experiments using neptune.ai
+    """
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        """Initialize the experiment tracker.
+
+        Args:
+            *args: Variable length argument list.
+            **kwargs: Arbitrary keyword arguments.
+        """
+        super().__init__(*args, **kwargs)
+        self.run_state: RunProvider = RunProvider()
+
+    @property
+    def config(self) -> NeptuneExperimentTrackerConfig:
+        """Returns the `NeptuneExperimentTrackerConfig` config.
+
+        Returns:
+            The configuration.
+        """
+        return cast(NeptuneExperimentTrackerConfig, self._config)
+
+    @property
+    def settings_class(self) -> Optional[Type["BaseSettings"]]:
+        """Settings class for the Neptune experiment tracker.
+
+        Returns:
+            The settings class.
+        """
+        return NeptuneExperimentTrackerSettings
+
+    def prepare_step_run(self, info: "StepRunInfo") -> None:
```

Comment:
I would suggest also overriding `cleanup_step_run(...)` to cleanup the run state. While this might not be strictly necessary as later steps will set the new state again, I think it's better to clean up this global state in order to avoid any issues.

Reply:
Ok, I'll see how to go about that, thanks

Reply:
Is there any way to know whether a given step is the last step of the pipeline it is in?

Reply:
Hang on. I think I got this :D

Reply:
Interesting logic, I think it should work :)

However there might be a more general flaw to it, but that depends on the logic that happens during the `run.sync()` and `run.stop()` calls.

When running a ZenML pipeline with a remote orchestrator, each step of the pipeline get's executed in a separate Docker container, which means that they won't be able to share any global state like the run state and also probably need to sync/stop the run separately?